From 9146e8c8ab963a50dec3142678be41d2fdf39d3f Mon Sep 17 00:00:00 2001
From: Jasper St. Pierre <jstpierre@mecheye.net>
Date: Sat, 2 Feb 2013 04:56:39 -0500
Subject: [PATCH] valagirparser: Ignore the new instance-parameter tag

g-ir-scanner will now start emitting the instance parameter; since
vala doesn't need this, just skip it.
---
 vala/valagirparser.vala |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/vala/valagirparser.vala b/vala/valagirparser.vala
index 0a90886..8256ef2 100644
--- a/vala/valagirparser.vala
+++ b/vala/valagirparser.vala
@@ -2896,6 +2896,11 @@ public class Vala.GirParser : CodeVisitor {
 			next ();
 
 			while (current_token == MarkupTokenType.START_ELEMENT) {
+				if (reader.name == "instance-parameter") {
+					skip_element ();
+					continue;
+				}
+
 				if (!push_metadata ()) {
 					skip_element ();
 					continue;
-- 
1.7.1

